app/controllers/application_controller.rb

```diff
@@ -78,47 +64,31 @@ class Base < Sinatra::Base
       settings.http_allow_headers.join(',')
   end
 
-  get '/ping' do
-    { 
-      status: 'Ok'
-    }.to_json
-  end
-
-  get '/jobs' do
-    jobs = []
-    etcd.get('/queue', recursive: true).children.each do |job|
-      job = JSON.parse(job.value)
-      jobs << JobPresenter.single(job) if job['created_from'] == 'API'
-    end
-    jobs.to_json
+  before do
+    Tendrl.etcd = Etcd.client(
+      host: settings.etcd_config[:host],
+      port: settings.etcd_config[:port],
+      user_name: settings.etcd_config[:user_name],
+      password: settings.etcd_config[:password]
+    )
   end
 
-  get '/jobs/:job_id' do
-    jobs = []
-    job = JSON.parse(etcd.get("/queue/#{params[:job_id]}").value)
-    JobPresenter.single(job).to_json
-  end
+  protected
 
-  get '/jobs/:job_id/logs' do
-    params[:type] ||= 'all'
-    job = JSON.parse(etcd.get("/queue/#{params[:job_id]}").value)
-    request_id = job['request_id']
-    logs = etcd.get("/#{request_id}/#{params[:type]}").value
-    { logs: logs, type: params[:type] }.to_json
+  def username
+    request.user_agent
```

Comment:
I think it is not good idea to use user agent for transferring user name because user agent header is shared among Web browsers and I expect that this API will be used also from UI.
As you can see here Github doesn't use User-Agent header for transferring user names: https://developer.github.com/v3/